packages/hash/subgraph/src/stdlib/element/entity.ts

```diff
@@ -0,0 +1,137 @@
+import { Subgraph } from "../../types/subgraph";
+import {
+  EntityEditionId,
+  EntityId,
+  isEntityEditionId,
+} from "../../types/identifier";
+import { isEntityVertex } from "../../types/vertex";
+import { Entity } from "../../types/element";
+
+/**
+ * Returns all `Entity`s within the vertices of the subgraph
+ *
+ * @param subgraph
+ */
+export const getEntities = (subgraph: Subgraph): Entity[] => {
+  return Object.values(
+    Object.values(subgraph.vertices).flatMap((versionObject) =>
+      Object.values(versionObject)
+        .filter(isEntityVertex)
+        .map((vertex) => vertex.inner),
+    ),
+  );
+};
+
+/**
+ * Gets an `Entity` by its `EntityEditionId` from within the vertices of the subgraph. Returns `undefined`
+ * if the entity couldn't be found.
+ *
+ * @param subgraph
+ * @param entityEditionId
+ * @throws if the vertex isn't an `EntityVertex`
+ */
+export const getEntityByEditionId = (
+  subgraph: Subgraph,
+  entityEditionId: EntityEditionId,
+): Entity | undefined => {
+  const [entityId, version] = [entityEditionId.baseId, entityEditionId.version];
+  const vertex = subgraph.vertices[entityId]?.[version];
+
+  if (!vertex) {
+    return undefined;
+  }
+
+  return vertex.inner;
+};
+
+/**
+ * Returns all `Entity`s within the vertices of the subgraph that match a given `EntityId`
+ *
+ * @param subgraph
+ * @param entityId
+ */
+export const getEntityEditionsByEntityId = (
+  subgraph: Subgraph,
+  entityId: EntityId,
+): Entity[] => {
+  const versionObject = subgraph.vertices[entityId];
+
+  if (!versionObject) {
+    return [];
+  }
+  const entityVertices = Object.values(versionObject);
+
+  return entityVertices.map((vertex) => {
+    return vertex.inner;
+  });
+};
+
+/**
+ * Gets an `EntityWithMetadata` by its `EntityId` whose lifespan overlaps a given `Date` moment
+ *
+ * @param subgraph
+ * @param entityId
+ * @param {string} timestamp An ISO-formatted datetime string of the moment to search for
+ *
+ * @throws if the vertices pointed to by `entityId` aren't `EntityVertex`es
```

Comment:
Ah this was from before I made `isEntityVertex` redundant in most cases. I wonder what we want to do here, perhaps we should return undefined if `!entityEditions`. I think I was treating it like a bug because it seemed like the subgraph was malformed if the entityId didn't exist at all within it.. but it's a bit strange to return `undefined` for the timestamp but throw if the entityId doesn't exist